docs/source/cli-reference/core-options.rst

```diff
@@ -119,3 +119,25 @@ Comparing Progress Message Options
     An example usage::
 
         scancode -clieu --json-pp sample.json samples --max-in-memory -1
+
+----
+
+``--max_depth`` Option
+----------------------------------
+
+    Normally, the scan takes place upto the maximum level of nesting of directories possible. But
+    using the ``--max-depth`` option, the depth of the scan can be limited. This can reduce the time
+    taken for the scan when deeper directories are not relevant.
+
+    Note that the ``--max-depth`` option will be ignored if you are scanning from a JSON file using
+    the ``--from-json`` option. In that case, the original depth is used.
+
+    Some INTEGER values of the ``--max-depth INTEGER`` option:
+
+    - **4**     - Scan only upto 4 levels of subdirectories
+    - **-1**    - Default, Scan upto maximum levels of subdirectories (no depth limit)
+    - **0**     - No files are scanned at all
```

Comment:
What about making 0 the default instead?  I am not sure we would ever want "No files are scanned at all"? and it feels better than -1 may be?

Reply:
Yeah, I think that makes much more sense. We can have positive integers for fixed depth while negative and zero for no limit and set 0 as the default.